index.js

```diff
@@ -61,13 +61,6 @@ function magnetURIDecode (uri) {
     }
   })
 
-  if (result['x.pe']) {
-    // remove duplicates and convert to array
-    result['x.pe'] = Array.isArray(result['x.pe'])
-      ? Array.from(new Set(result['x.pe']))
-      : [result['x.pe']]
-  }
-
```

Comment:
The code for duplicate removal should only run for the friendly API -- `peerAddresses` -- just like we run it for `announce` but we leave `tr` as the raw value.